src/components/MultiSelect/chips.js

```diff
@@ -3,19 +3,24 @@ import PropTypes from 'prop-types';
 import { StyledChip } from './styled';
 
 function Chips(props) {
-    const { value, variant, onDelete } = props;
+    const { value, variant, onDelete, disabled, readOnly } = props;
     if (!value) {
         return null;
     }
+
     if (Array.isArray(value)) {
-        return value.map(val => (
-            <StyledChip
-                key={val.name}
-                label={val.label}
-                variant={variant}
-                onDelete={() => onDelete(val)}
-            />
-        ));
+        return value.map(val => {
+            const onDeleteCallback = disabled || readOnly ? null : () => onDelete(val);
```

Comment:
I think this logic works best in the handleDelete declaration in the parent component. It is only once and you don't have to be passing disable and readonly to this component.

Reply:
Each chip should call handleDelete with its own value, and the only way of doing it that occurred to me is this way, I will be thankful if you could explain to me how do you think it will be better

Reply:
@HellWolf93 I think whats @yvmunayev meant is to use this logic in parent component and then pass below a prop which could be null or a function then here you not need to use disabled and readOnly and also repeat the logic